### ISSUE TITLE
[Bug]: Git | Unable to connect to Git with root SSH

### ISSUE THREAD

**Issue by Aishwarya-U-R** [appsmithorg/appsmith#20400]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

<img width="1531" alt="Screenshot 2023-02-06 at 3 13 08 PM" src="https://user-images.githubusercontent.com/91450662/216938425-5b011998-a7a6-4b59-b1a3-ec54a9d9b25c.png">

### Steps To Reproduce

1. Pull TED - which hosts Gitea at 3000
2. Open Appsmith app & connect to Git via - `ssh://root@host.docker.internal:Cypress/123dsdsf.git`
3. Generate Key & add it to the Gitea Deploy keys into this Repo
4. Click Connect from Appsmith
5. **Observe: GIT ACTION EXECUTION ERROR
Git command execution error: Exception caught during execution of fetch command**
6. **Expected: The Git connection should work fine & user should be able to proceed**

### Version

Cloud/Release v1.9.6

**Comment by AnaghHegde** [comment]
@Aishwarya-U-R Is this an issue you are able to reproduce locally? If it is intermittent then can we reduce the priority?

**Comment by Aishwarya-U-R** [comment]
Yes it is occurring & always. We are testing it locally & unless this passes - we cant run it in CI.

**Comment by AnaghHegde** [comment]
The connection URL `ssh://root@host.docker.internal:Cypress/123dsdsf.git` is this the actual git repo url? I am not sure if the jgit supports redirections for a ssh command.

**Comment by Aishwarya-U-R** [comment]
@AnaghHegde - Yes it is actual url for gitea ssh that is installed & running as part of TED locally.
cc: @yatinappsmith

**Comment by AnaghHegde** [comment]
@Aishwarya-U-R What error do you seen when you try to ssh from the terminal from your local machine?

If the gitea is running inside the TED, is the PORT 22 available? Have you verified this?
